src/solver/ts-generator/thermal.cpp

```diff
@@ -60,7 +60,7 @@ ThermalInterface::ThermalInterface(Data::AreaLink::LinkTsGeneration& source,
  plannedVolatility(source.plannedVolatility),
  forcedLaw(source.forcedLaw),
  plannedLaw(source.plannedLaw),
- prepro(source.prepro),
+ prepro(source.prepro.get()),
```

Comment:
Maybe already said in the review of [PR 1986](https://github.com/AntaresSimulatorTeam/Antares_Simulator/pull/1986), but thermal.cpp is a bad name for this file.
It mostly contains code about TS generation (and a certain kind of TS generation).
So we should rename it for example **availabilityTSgeneration.cpp** (maybe not the right name, by the main idea is here).
The code unrelated to TS generation (for example code specific to links or thermal clusters) should be moved elsewhere

Reply:
Not taken care of yet. Is there a problem to perform this change ?

Reply:
[done](https://github.com/AntaresSimulatorTeam/Antares_Simulator/pull/2019/commits/b647031d0cf14beedfc0ac21afe6fdd5ef302c57)